build-llvm34.md

```diff
@@ -70,77 +63,130 @@ If you want to build KLEE with LLVM 2.9, [click here]({{site.baseurl}}/build-llv
+
+6. **(Optional) Install lit:**
+
+   For testing the `lit` tool is used. If you LLVM from a build tree you can
+   skip this step as the build system will try to use `llvm-lit` in the
+   directory containing the LLVM binaries.
+
+   If you don't want to run the tests you can skip this step but you will need
+   to pass `-DENABLE_TESTS=OFF` to CMake when configuring KLEE in step 9.
 
-   Build Google test libraries for unit tests. We do a manual build, because the libgtest-dev package (version 1.6) installed through apt does not work for us.  
+   ```bash
+   $ pip install lit
+   ```
+
+7. **(Optional) Install tcmalloc:**
+
+   By default, KLEE uses `malloc_info()` to observe and to restrict its memory usage.
+   Due to limitations of `malloc_info()`, the maximum limit is set to 2 GB. To support bigger limits, KLEE can use TCMalloc as an alternative allocator. It is thus necessary to install TCMalloc:
 
    ```bash
-   $ curl -OL https://googletest.googlecode.com/files/gtest-1.7.0.zip  
-   $ unzip gtest-1.7.0.zip  
-   $ cd gtest-1.7.0  
-   $ cmake .  
-   $ make  
-   $ cd ..
+   $ sudo apt-get install libtcmalloc-minimal4 libgoogle-perftools-dev
    ```
 
-6. **Get KLEE source:**  
+   When configuring KLEE in step 9 pass `-DENABLE_TCMALLOC=ON` to CMake when configuring KLEE.
+
+8. **Get KLEE source:**  
 
    ```bash
    $ git clone https://github.com/klee/klee.git
    ```
 
-7. **Configure KLEE:** From the KLEE source directory, run:  
+9. **Configure KLEE:**
+
+   KLEE must be built "out of source" so first make a binary build directory. You can create this where ever you like.
 
    ```bash
-   $ ./configure --with-stp=/full/path/to/stp/build --with-uclibc=/full/path/to/klee-uclibc --enable-posix-runtime
+   $ mkdir klee_build_dir
    ```
 
-   **NOTE:** If LLVM is not found or you have multiple LLVM versions installed, you can add `--with-llvmsrc=/usr/lib/llvm-3.4/build --with-llvmobj=/usr/lib/llvm-3.4/build --with-llvmcc=/usr/bin/clang-3.4 --with-llvmcxx=/usr/bin/clang++-3.4`.  
-If you skipped step 4, simply remove the `--with-uclibc` and `--enable-posix-runtime` options.<br/><br/>  
+   Now `cd` into the build directory and run CMake to configure KLEE where `<KLEE_SRC_DIRECTORY>` is the path
+   to the KLEE git repository you cloned in step 7.
 
-8. **Build KLEE:**  
+   ```bash
+   $ cd klee_build_dir
+   $ cmake <CMAKE_OPTIONS> <KLEE_SRC_DIRECTORY>
+   ```
+
+   `<CMAKE_OPTIONS>` are the configuration options. These are documented in [README-CMake.md](https://github.com/klee/klee/blob/master/README-CMake.md).
+
+   For example if KLEE was being built with STP, the POSIX runtime, klee-uclibc and testing then the
+   command line would look something like this
 
    ```bash
-   $ make  
+   cmake \
+     -DENABLE_SOLVER_STP=ON \
+     -DENABLE_POSIX_RUNTIME=ON \
+     -DENABLE_KLEE_UCLIBC \
+     -DKLEE_UCLIBC_PATH=<KLEE_UCLIBC_SOURCE_DIR> \
+     -DGTEST_SRC_DIR=<GTEST_SOURCE_DIR> \
+     -DENABLE_TESTS=ON \
```

Comment:
(Typo fixed).  OK, glad we're on the same page here.  OK, let's change it later, as I really think they're complicating things.